todo/src/components/TodoForm.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { addTodo, deleteCompleted } from '../actions/actions';
+
+class TodoForm extends React.Component {
+  state = { todo: '' };
+
+  changeHandler = e => {
+    e.preventDefault();
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  submitTodo = e => {
+    e.preventDefault();
+    this.props.addTodo(this.state.todo);
+    this.setState({
+      todo: ''
+    });
+  };
+
+  render() {
+    return (
+      <form onSubmit={e => this.submitTodo(e)}>
```

Comment:
note that there is an implicit event being passed in with this synthetic event listener, so you could just write {this.submitTodo}